sound/soc/sof/pcm.c

```diff
@@ -687,6 +687,37 @@ static int sof_pcm_dai_link_fixup(struct snd_soc_pcm_runtime *rtd,
 	return 0;
 }
 
+static void sof_set_longname(struct snd_soc_component *component)
+{
+	struct snd_sof_dev *sdev = snd_soc_component_get_drvdata(component);
+	struct snd_sof_pdata *pdata = sdev->pdata;
+	char flavour[80];
+	int pos = 0;
+	int i;
+
+	pos += sprintf(flavour, "%s", "SOF");
```

Comment:
You could use snprintf() in the same way as snd_soc_set_dmi_name() does. Granted the strings you add are fixed length, but this would make it more future proof for any modifications of the string syntax.

Reply:
yes definitely, just a quick thing to get something out, also to clear things for myself...